parameter file added to request dsl

bimanaev-fix opened the pull request:
Добавлена возможность загрузки файлов используя request dsl:
file(filename, stream)

bimanaev-fix committed:
parameter file added to request dsl

swarmshine commented on kbdd/src/test/kotlin/ru/fix/kbdd/rest/RestTest.kt:
```diff
@@ -392,4 +393,31 @@ class RestTest {
+            headers { "my-header" % "header-value" }
+            baseUri(server.baseUrl())
+            post("/post-form-data-request")
+            file("resource-with-placeholders.json", File("src/test/resources/resource-with-placeholders.json").inputStream())
```
File tries to open resource based on current working directory. Better to use 
`RestTest::class.java.getResourceAsStream("/resource-with-placeholders.json")` this way we are  looking in test resource independently on build environment settings.

swarmshine commented on kbdd/src/test/kotlin/ru/fix/kbdd/rest/RestTest.kt:
```diff
@@ -392,4 +393,31 @@ class RestTest {
+            baseUri(server.baseUrl())
+            post("/post-form-data-request")
+            file("resource-with-placeholders.json", File("src/test/resources/resource-with-placeholders.json").inputStream())
+        }
```
resource-with-placeholders.json is used by another test, it content can be changed or removed. 
Better to use new resource for new test. 
mulipart-data-file.json for example with some test content

swarmshine commented on kbdd/src/test/kotlin/ru/fix/kbdd/rest/RestTest.kt:
```diff
@@ -392,4 +393,31 @@ class RestTest {
+                                        .withHeader("Content-Disposition", containing("form-data"))
+                                        .withHeader("Content-Disposition", containing("name=\"file\""))
+                                        .withHeader("Content-Disposition", containing("filename=\"resource-with-placeholders.json\""))
+                                        .withHeader("Content-Type", equalTo("application/octet-stream"))
```
could you please add verification that actual content of the file was sent, like: 
`.withBody(containing("file content here"))`

bimanaev-fix committed:
test updated:
1) multipart body check added
2) using file changed
3) file.inputStream to class.getResourceAsStream
